### PR TITLE
Jetpack connect: Separate reducers

### PR DESCRIPTION
**Pull request by sirreal**
A large state tree was represented in a single file at `jetpack-connect/reducer`. This PR separates several sub-trees into separate files and uses an index to combine them.

The tests are similarly separated. This helps to clarify the boundaries of the different sub-trees and will simplify refactor as we attempt to move away from one large stateful operation towards stateless services and being to prune this tree.

There are no behavioral changes.

## Testing:

* Notice that the first commit (617ad1b) includes a simple snapshot of the initial state tree, which is moved and renamed as necessary in the following commit (20041f4), but is otherwise unchanged throughout the history of this PR. That provides a high level of confidence that these Janitorial changes have left the state tree functioning as before.

* Test the different connection flows that depend on this tree, especially connection and SSO.

### CODE REVIEW

File: client/state/jetpack-connect/reducer/test/jetpack-connect-site.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+		const state = jetpackConnectSite(
+			{ url: 'https://automattic.com' },
+			{
+				type: JETPACK_CONNECT_CHECK_URL_RECEIVE,
+				url: 'https://example.wordpress.com',
+				data: data,
+			}
+		);
+
+		expect( state ).to.eql( { url: 'https://automattic.com' } );
```
Comment (seear) [inline client/state/jetpack-connect/reducer/test/jetpack-connect-site.js]: missing a check on `isFetched` according to the name of the test?

Comment (sirreal) [reply client/state/jetpack-connect/reducer/test/jetpack-connect-site.js]: 👍  added a test in c0e3e81d4a91981f43f3c7b98ae562f49da0a969

File: client/state/jetpack-connect/reducer/test/jetpack-connect-authorize.js [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+			type: JETPACK_CONNECT_AUTHORIZE_RECEIVE,
+			data,
+		} );
+
+		expect( state ).to.have.property( 'authorizeError' ).to.be.false;
+		expect( state ).to.have.property( 'authorizeSuccess' ).to.be.true;
+		expect( state ).to.have.property( 'autoAuthorize' ).to.be.false;
+		expect( state )
+			.to.have.property( 'plansUrl' )
+			.to.eql( data.plans_url );
```
[the remark this answers is not in the document]
Comment (sirreal) [reply client/state/jetpack-connect/reducer/test/jetpack-connect-authorize.js]: Get ready for lots of that 😀

**Comment by sirreal** [comment]
Thanks for the lengthy and in depth review!

**Pull request merged**